src/api/app/models/package.rb

```diff
@@ -313,8 +313,13 @@ def master_product_object
     product_object || self
   end
 
-  def check_write_access(ignoreLock = nil)
-    User.current.can_modify_package? master_product_object, ignoreLock
+  def check_write_access(ignoreLock = nil, user = User.current)
+    user.can_modify_package? master_product_object, ignoreLock
+  end
+
+  # For the sake of having a more speaking method name...
+  def can_be_modified_by?(user)
+    check_write_access(nil, user)
```

Comment:
We could even replace check_write_access completely. It's only used 3 times in OBS.